scripts/what_changed.py

```diff
@@ -0,0 +1,341 @@
+#!/usr/bin/env python3
+
+# A script to generate a list of boards that have changed or added and create an
+# arguemnts file for sanitycheck to allow running more tests for those boards.
+
+import sys
+import re, os
+from email.utils import parseaddr
+import sh
+import argparse
+from github import Github
+
+if "ZEPHYR_BASE" not in os.environ:
+    exit(1)
+
+repository_path = os.environ['ZEPHYR_BASE']
+sh_special_args = {
+    '_tty_out': False,
+    '_cwd': repository_path
+}
+
+def git(*args):
+    # Runs a git command using the 'sh' library (https://amoffat.github.io/sh/)
+
+    # Do not create a TTY for stdout, so that Git doesn't start a pager.
+    #
+    # Hack: Setting _cwd to the working dir seems pointless as of writing (it
+    # should be the default), but keep it around in case it's working around
+    # some issue
+    return sh.git(*args, _tty_out=False, _cwd=os.getcwd())
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+                description="Generate a sanitycheck argument for for boards "
+                            " that have changed")
+    parser.add_argument('-c', '--commits', default=None,
+            help="Commit range in the form: a..b")
+    parser.add_argument('-r', '--repo', default=None,
+                        help="Github repository")
+    parser.add_argument('-p', '--pull-request', default=0, type=int,
+                        help="Pull request number")
+
+    return parser.parse_args()
+
+def main():
+    boards = set()
+
+    args = parse_args()
+    if not args.commits:
+        exit(1)
+
+    commit = git("diff","--name-only", args.commits)
+    files = commit.split("\n")
+
+    docs = 0
+    images = 0
+    kernel = 0
+    code = 0
+    net = 0
+    bt = 0
+
+    matches = [
```

Comment:
Is it possible that we encode the 'matches' as a file in the gitrepo similar to CODEOWNERS or MAINTAINER file syntax so we can update it as new code area/labels are created?

Reply:
yeah, thought about it, I now want to get this in, get some feedback, and optimise. But a label file similar to CODEOWNER is definitely on the plan.

I am also looking for a way to identify complexity of changes and set either Maintainer or Trivial labels.

Reply:
Agreed, having something working first will be great.  Can't wait for API, DTS labels to get automatically set!!

Reply:
Agreed, let's have a second pass with the regexes and labels moved to a separate file